model/AbstractRdsResultStorage.php

```diff
@@ -0,0 +1,727 @@
+            $persistenceId = $this->hasOption(self::OPTION_PERSISTENCE) ?
+                $this->getOption(self::OPTION_PERSISTENCE)
+                : 'default';
+            $this->persistence = $this->getServiceLocator()->get(PersistenceManager::SERVICE_ID)->getPersistenceById($persistenceId);
+        }
+
+        return $this->persistence;
+    }
+
+    /**
+     * @param $value
+     *
+     * @return mixed
+     */
+    protected function unserializeVariableValue($value)
+    {
+        return unserialize($value);
+    }
+
+    /**
+     * @param $value
+     *
+     * @return string
+     */
+    protected function serializeVariableValue($value)
+    {
+        return serialize($value);
+    }
+
+    /**
+     * @param array $data
+     *
+     * @throws DuplicateVariableException
+     */
+    private function insertMultiple(array $data)
+    {
+        $duplicatedData = false;
+        try {
+            $this->getPersistence()->insertMultiple(self::VARIABLES_TABLENAME, $data);
+        } catch (UniqueConstraintViolationException $e) {
+            $duplicatedData = true;
+            foreach ($data as $row) {
+                try {
+                    $this->getPersistence()->insert(self::VARIABLES_TABLENAME, $row);
+                } catch (UniqueConstraintViolationException $e) {
+                    //do nothing, just skip it
+                }
+            }
+        }
+
+        if ($duplicatedData) {
+            throw new DuplicateVariableException(sprintf('An identical result variable already exists.'));
+        }
+    }
+
+    public function spawnResult()
+    {
+        \common_Logger::w('Unsupported function');
+    }
+
+    /*
+     * retrieve specific parameters from the resultserver to configure the storage
+     */
+    public function configure($callOptions = [])
+    {
+        \common_Logger::d('configure  RdsResultStorage with options : ' . implode(" ", $callOptions));
+    }
+
+    /**
+     *
+     * @param mixed $a
+     * @param mixed $b
+     *
+     * @return number
+     */
+    public static function sortTimeStamps($a, $b)
+    {
+        list($usec, $sec) = explode(" ", $a);
+        $floata = ((float)$usec + (float)$sec);
+        list($usec, $sec) = explode(" ", $b);
+        $floatb = ((float)$usec + (float)$sec);
+        //common_Logger::i($a." ".$floata);
+        //common_Logger::i($b. " ".$floatb);
+        //the callback is expecting an int returned, for the case where the difference is of less than a second
+        //intval(round(floatval($b) - floatval($a),1, PHP_ROUND_HALF_EVEN));
+        if ((floatval($floata) - floatval($floatb)) > 0) {
+            return 1;
+        } elseif ((floatval($floata) - floatval($floatb)) < 0) {
+            return -1;
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * Creates the table for results storage.
+     *
+     * @param Schema $schema
+     *
+     * @return Table
+     * @throws SchemaException
```

Comment:
are you sure that this is thrown ?

Reply:
Yes, it can be by Schema::createTable method.